CoreGraphUtilities/src/au/gov/asd/tac/constellation/graph/utilities/hashmod/HashmodAction.java

```diff
@@ -65,14 +67,23 @@ public void actionPerformed(final ActionEvent ev) {
         final DialogDescriptor dialog = new DialogDescriptor(hashmodPanel, Bundle.MSG_Title(), true, e -> {
             if (e.getActionCommand().equals("OK")) {
                 final Hashmod hashmod1 = hashmodPanel.getHashmod();
+                final Boolean isChainedHashmods = hashmodPanel.isChainedHashmods();
+                final boolean createAttributes = hashmodPanel.getCreateAttributes();
+                final Hashmod[] chainedHashmods = hashmodPanel.getChainedHashmods();
+                final int numChainedHashmods = hashmodPanel.numChainedHashmods();
                 final Boolean createNonMatchingKeysVertexes = hashmodPanel.getCreateVertexes();
                 hashmodPanel.setAttributeNames(hashmod1.getCSVKey(), hashmod1.getCSVHeader(1), hashmod1.getCSVHeader(2));
 
                 PluginExecution.withPlugin(new SimpleEditPlugin(Bundle.CTL_HashmodAction()) {
                     @Override
                     public void edit(final GraphWriteMethods wg, final PluginInteraction interaction, final PluginParameters parameters) throws InterruptedException {
                         if (hashmod1 != null) {
-                            HashmodAction.run(wg, interaction, hashmod1, createNonMatchingKeysVertexes);
+                            HashmodAction.run(wg, interaction, hashmod1, createNonMatchingKeysVertexes, true, createAttributes);
+                        }
+                        if (isChainedHashmods && numChainedHashmods >= 2) {
+                            for (int i = 1; i < numChainedHashmods; i++) {
```

Comment:
Does the for loop need to run over the first chained hashmod in the array? If so, you'll need to start at i=0

Reply:
The first one (hashmod1) has already been run so it starts from the second (aka chainedHashmods[i])

Reply:
I figured it would be something like that, just wanted to make sure that was intentional :)